src/media/playback/playback_device.cpp

```diff
@@ -477,22 +510,41 @@ void playback_device::do_loop(T action)
         }
 
         //On failure, exit thread
-        if(action_succeeded == false)
+        if(action_succeeded == false && m_is_started)
         {
-            for (auto s : m_active_sensors)
-                s.second->flush_pending_frames();
-
-            //Go over the sensors and stop them
-            size_t active_sensors_count = m_active_sensors.size();
-            for (size_t i = 0; i<active_sensors_count; i++)
+            std::vector<std::shared_ptr<playback_sensor>> playback_sensors_copy;
```

Comment:
Add comment why do we need local copy of playback_sensors

Reply:
```
// Stopping the sensor will call another function which will remove the sensor from the
// list of active sensors, which will cause issues -- so we copy it first
```